Assets/Scripts/PlantSystem/Venus.cs

```diff
@@ -1,30 +1,37 @@
 // TODO make breakable pod
 
+using System;
 using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.Serialization;
 
 public class Venus : CursedBehaviour {
+    private Damageable damageable;
     private EyeSight eyesight;
     public GameObject display;
 
     protected override void ExtraAwake() {
+        damageable = GetComponent<Damageable>();
         eyesight = GetComponent<EyeSight>();
         display.SetActive(false);
 
         eyesight.enabled = false;
     }
 
     public void OnSeeingObject(GameObject objectToEat) {
-        if (objectToEat == GlobalRoomState.player) {
-            Destroy(objectToEat);
-            return;
-        }
-
-        if (objectToEat.GetComponent<ZombieBehaviour>() != null) {
-            Destroy(objectToEat);
-            Destroy(gameObject);
+        if (objectToEat.TryGetComponent(out Edible edible)) {
```

Comment:
```suggestion
        if (!objectToEat.TryGetComponent(out Edible edible)) {
            return;
        }
```

Convert this into an early return instead